File: src/main/java/mx/edu/unpa/adoptame/service/MascotaService.java
package mx.edu.unpa.adoptame.service;

import java.util.Optional;

import mx.edu.unpa.adoptame.domain.Mascota;


public interface MascotaService {
	
		public Iterable<Mascota> findAll();
		public Iterable<Mascota> findByUsuarioId(Integer idUsuario);
		public Optional<Mascota> findById(Integer id);
		public Optional<Mascota> findByName(String nombre);
		public Mascota save(Mascota mascota);
		public Mascota updateEstadoAdopcion(Integer idMascota, String estadoAdopcion, Integer idUsuario);
		public void deleteById(Integer id);
	}

File: app/src/main/java/mx/edu/unpa/adoptame/service/MascotaService.kt
package mx.edu.unpa.adoptame.service

import mx.edu.unpa.adoptame.model.Mascota
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface MascotaService {
    @GET("mascota")
    fun getMascotas(): Call<List<Mascota>>

    @GET("mascota/{idMascota}")
    fun getMascota(@Path("idMascota") idMascota: Int): Call<Mascota>

    @GET("mascota/name")
    fun getMascotaByName(@Query("name") name: String): Call<Mascota>

    @POST("mascota")
    fun crearMascota(@Body mascota: Mascota): Call<Mascota>

    @GET("mascota/usuario/{idUsuario}")
    fun getMascotasByUsuario(@Path("idUsuario") idUsuario: Int): Call<List<Mascota>>

    @PATCH("mascota/{idMascota}/estado")
    fun actualizarEstado(
        @Path("idMascota") idMascota: Int,
        @Body body: Map<String, @JvmSuppressWildcards Any>
    ): Call<Mascota>
}
